src/modules/launcher/Plugins/Microsoft.Plugin.Program/Programs/Win32.cs

```diff
@@ -468,9 +468,9 @@ public int GetHashCode(Win32 obj)
                 int fullPathPrime = 31;
 
                 int result = 1;
-                result = result * namePrime + obj.Name.GetHashCode();
-                result = result * executablePrime + obj.ExecutableName.GetHashCode();
-                result = result * fullPathPrime + obj.FullPath.GetHashCode();
+                result = result * namePrime + obj.Name.ToLowerInvariant().GetHashCode();

```

Comment:
Is there any reason to use `ToLowerInvariant` as compared to `ToLower`? My understanding is that in this scenario we would need a culture-specific lowercase letter because `ExecutableName` and `Name` could be localized.

Reply:
It is precisely to ignore any culture differences in `Name` that I have used ToLowerInvariant(). We would not want the same application to show up in multiple languages as it would still be a duplicate. It is recommended to use ToLower only when the input of the user in required in a particular culture.

Reference - https://stackoverflow.com/questions/6225808/string-tolower-and-string-tolowerinvariant.

Reply:
If you need the lowercase or uppercase version of an operating system identifier, such as a file name, named pipe, or registry key, use the ToLowerInvariant or ToUpperInvariant method. This produces the same result in every culture and performs more efficiently.

Ref - https://docs.microsoft.com/en-us/dotnet/api/system.string.tolower?view=netcore-3.1

Reply:
Using `ToLowerInvariant()` will fail if the strings are localized in Turkish. I found this issue mentioned in the last answer on this blog post: https://stackoverflow.com/questions/6225808/string-tolower-and-string-tolowerinvariant. You can find a sample code below: 

`input` and `input2` both represent the same word in lowercase and uppercase respectively. Using `ToLower` would give the correct results but using `ToLowerInvariant` gives incorrect results. 
```
Thread.CurrentThread.CurrentCulture = new CultureInfo("tr-TR"); 
const string input = "ia";
const string input2 = "İA";
bool comparison = input.ToLowerInvariant().GetHashCode() == input2.ToLowerInvariant().GetHashCode();
Console.WriteLine("These things are equal: " + comparison);
```

Reply:
Yes, that seems to be the exception only for that character `I` of the turkish language, for using ToLowerInvariant(). However, if we use ToLower(), we would be missing out on all the other cases. We do not have the functionality to use `OrdinalCase` in ToLower(CultureInfo), therefore this is the best we can do. I have used OrdinalCase to compare the equality of strings while overloading the Equals function. As there is nothing else that we can do because this is inherently present in ToLowerInvariant(), there may be duplicates for the letter i of the turkush language. If you have a better solution to this please let me know. If not, I shall go ahead and merge it.

Reply:
I didn't find anything that might solve it except using `toLower` but that might not be applicable to a lot more cases as you mentioned. So, please go ahead and merge this.